server/blob/blob-cloud/src/test/java/org/apache/james/blob/cloud/CloudBlobsDAOTest.java

```diff
@@ -0,0 +1,91 @@
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.blob.cloud;
+
+import java.net.URI;
+import java.util.Properties;
+import java.util.UUID;
+
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.HashBlobId;
+import org.apache.james.blob.api.ObjectStore;
+import org.apache.james.blob.api.ObjectStoreContract;
+import org.jclouds.ContextBuilder;
+import org.jclouds.blobstore.BlobStore;
+import org.jclouds.blobstore.BlobStoreContext;
+import org.jclouds.openstack.keystone.config.KeystoneProperties;
+import org.jclouds.openstack.swift.v1.reference.TempAuthHeaders;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(DockerSwiftTempAuthExtension.class)
+public class CloudBlobsDAOTest implements ObjectStoreContract {
+    private static final String TENANT_NAME = "test";
+    private static final String USER_NAME = "tester";
+    private static final Credentials PASSWORD = Credentials.of("testing");
+    private static final Identity IDENTITY = Identity.of(TENANT_NAME + ":" + USER_NAME);
+
+    private URI swiftEndpoint;
+    private ContainerName containerName;
+    private BlobStore blobStore;
+
+    @BeforeEach
+    void setUp(DockerSwiftTempAuthExtension.DockerSwift dockerSwift) throws Exception {
+        final Properties overrides = new Properties();
+        overrides.setProperty(KeystoneProperties.CREDENTIAL_TYPE, "tempAuthCredentials");
+        overrides.setProperty(TempAuthHeaders.TEMP_AUTH_HEADER_USER, "X-Storage-User");
+        overrides.setProperty(TempAuthHeaders.TEMP_AUTH_HEADER_PASS, "X-Storage-Pass");
+        swiftEndpoint = dockerSwift.getEndpoint();
+        BlobStoreContext blobStoreContext = ContextBuilder.newBuilder("openstack-swift")
+            .endpoint(swiftEndpoint.toString())
+            .credentials(IDENTITY.getValue(), PASSWORD.getValue())
+            .overrides(overrides)
+            .buildView(BlobStoreContext.class);
+        blobStore = blobStoreContext.getBlobStore();
+        containerName = ContainerName.of(UUID.randomUUID().toString());
+        blobStore.createContainerInLocation(null, containerName.getValue());
```

Comment:
don't we need a component that handles this container creation in the production code ?

Reply:
This area of the design is undefined/unclear for the moment. The container can be created and administered by ops externally from the application. 
There may be multiple different containers to store different parts (attachments, headers, bodies) or not. 
The integration design is not completely clear to me yet so I held off.